this is the first project

HermelaDemeke opened the pull request:
Surely ! This code is an example of a Java class called `Person' , 'Student' ,and 'Teacher'  that represents individuals and their attributes. Let's go through the code and explain its different category :

1. The class `Person` is defined with four instance variables: 'id', 'name', 'age', and 'parent_Permission`. These variables will store the unique identifier, name, age, and parent permission status of a person, respectively.

2. The class has two constructors. The first constructor takes in four parameters:  'id', 'name', 'age', and 'parent_Permission` It initializes the corresponding instance variables with the provided values.

3. The second constructor is an alternative to the first one, with only two parameters: `id` and `age`. It makes use of the first constructor by passing `id`, "Unknown" (as the default name value), `age`, and `true` (as the default parent permission value).

4. Getter methods (`getId()`, `getName()`, `getAge()`) are provided to access the private instance variables. These methods enable other parts of the program to retrieve the values of these variables.

5. Setter methods (`setName(String name)`, `setAge(int age)`) are provided to modify the values of the `name` and `age` variables.

6. The `isOfAge()` method is a private helper method to checks whether a person is of age 18. It returns a boolean value (true if the person is 18 or older, false otherwise).

7. The `canUseServices()` method determines if a person can use certain services. It returns true ifthe person is of (`isOfAge()` which means age 18 or older it returns true) or if they have parent permission (`parentPermission` is true).
*The other two classes are extended from the Person class(inherit from person) and a little parameters added to it ,like specialization and in the Teacher class there is an override in the method can_use_service. 
Finally each class saved in separate java file.

HermelaDemeke committed:
this is the first project

Benawi commented on Person.java:
```diff
@@ -0,0 +1,57 @@
+     name and parent_permission are optional and have default values of "Unknown" and true.
+     */
+    public Person(int age){
+        this.age=age;
```
- [ ] You do not need to have this constructor

Benawi commented on Person.java:
```diff
@@ -0,0 +1,57 @@
+        this.name="unknown";
+        this.parent_permission=true;
+    }
+    public Person(String name, int age, boolean parent_permission) {
```
public Person(String name='Unknown', int age, boolean parent_permission =true)

Benawi commented on Teacher.java:
```diff
@@ -0,0 +1,16 @@
+    //Constructor extends parent's constructor by adding specialization and a parameter for it.
+    public Teacher(String name, int age, boolean parent_permission) {
+        super(name, age, parent_permission);
+        this.specialization=specialization;
```
Specialization is not defined in public Teacher(String name, int age, boolean parent_permission)

Benawi reviewed (changes requested):
#### Hi @HermelaDemeke 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

![giphy](https://user-images.githubusercontent.com/25227022/169391888-8dd8108f-0d42-4e54-87cc-a3f9fd392dbc.gif)


## Highlights

✔️  Create Person Class 
✔️  Create Student Class 
✔️  Create Teacher Class 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. @benawi

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

HermelaDemeke committed:
Update Teacher.java

Specialization not defined in public Teacher(String name, int age, boolean parent_permission)

HermelaDemeke committed:
Update Person.java

do not need to have Public Person (int age) constructor and,
name and parent_permission have default value 'unknown' and 'true' simultaneously.

Benawi reviewed (approved):
Hi @HermelaDemeke 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 

![Alt Text](https://media.giphy.com/media/JoSiL3L4ON8DlyVxv0/giphy.gif)
## Highlights
✔️ Create Person Class
✔️ Create Student Class
✔️ Create Teacher Class
✔️ PR is well descriptive 👍 
✔️ Requested changes implemented 👍
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you  take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@benawi** in your question so I can receive the notification.

------
